core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -270,9 +270,25 @@ private[spark] object JettyUtils extends Logging {
 
       gzipHandlers.foreach(collection.addHandler)
       connectors.foreach(_.setHost(hostName))
+      // As each acceptor and each selector will use one thread, the number of threads should at
+      // least be the number of acceptors and selectors plus 1. (See SPARK-13776)
+      var minThreads = 1
+      connectors.foreach { c =>
+        // Currently we only use "SelectChannelConnector"
+        val connector = c.asInstanceOf[SelectChannelConnector]
+        // Limit the max acceptor number to 8 so that we don't waste a lot of threads
+        connector.setAcceptors(math.min(connector.getAcceptors, 8))
+        // The number of selectors always equals to the number of acceptors
+        minThreads += connector.getAcceptors * 2
+      }
       server.setConnectors(connectors.toArray)
 
       val pool = new QueuedThreadPool
+      if (serverName.nonEmpty) {
+        pool.setName(serverName)
+      }
+      pool.setMaxThreads(math.max(pool.getMaxThreads, minThreads))
+      pool.setMinThreads(math.min(pool.getMinThreads, pool.getMaxThreads))
```

Comment:
If `minThreads=4, getMaxThreads=100, getMinThreads=8`, I want to call `setMaxThreads(100), setMinThreads(8)`.
If `minThreads=200, getMaxThreads=100, getMinThreads=8`, I want to call `setMaxThreads(200), setMinThreads(8)`.
If `minThreads=10, getMaxThreads=100, getMinThreads=8`, I want to call `setMaxThreads(200), setMinThreads(8)`.